mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -680,8 +680,11 @@ void mlir::scf::promote(RewriterBase &rewriter, scf::ForallOp forallOp) {
   SmallVector<Value> results;
   results.reserve(forallOp.getResults().size());
   for (auto &yieldingOp : terminator.getYieldingOps()) {
+    // Skip non-ParallelInsertSliceOp operations
     auto parallelInsertSliceOp =
-        cast<tensor::ParallelInsertSliceOp>(yieldingOp);
+        dyn_cast<tensor::ParallelInsertSliceOp>(yieldingOp);
```

Comment:
Can we make this all work with the interface? Removing this dependency from SCF on the tensor dialect would be a very nice improvement.

Reply:
I believe this needs to be a per-op basis analysis.
One thing we can do is to have a `canPromote` interface method to make a decision here so each can implement its own decision.

Reply:
We will make this change so it can apply the promotion in a next patch.